utils/questions.py

```diff
@@ -1,10 +1,286 @@
+                }
+            }
+        }
+    '''
+    }
+
+    try:
+        response = requests.post(url, json=data, headers=headers, timeout=10)
+
+    except Exception as e:
+        logger.exception(
+            "file: embeds/question_embeds.py ~ Daily problem could not be retrieved: %s", e)
+
+        embed = error_embed("Daily problem could not be retrieved")
+        return embed
+
+    if response.status_code != 200:
+        logger.exception(
+            "file: embeds/question_embeds.py ~ Daily problem could not be retrieved. Error code: %s", response.status_code)
+
+        embed = error_embed("Daily problem could not be retrieved")
+        return embed
+
+    response_data = response.json()
+
+    question_title = response_data['data']['challenge']['question']['titleSlug']
+
+    return get_question_info_from_title(question_title)
+
+def get_question_info_from_title(question_title_slug):
+    logger.info(
+        "file: utils/questions.py ~ get_question_info_from_title ~ run")
+    
+    url = 'https://leetcode.com/graphql'
+
+    headers = {
+        'Content-Type': 'application/json',
+    }
+
+    # Get question info
+    data = {
+        'operationName': 'questionInfo',
+        'query':
+        '''
+        query questionInfo($titleSlug: String!) {
+            question(titleSlug: $titleSlug) {
+                questionFrontendId
+                title
+                difficulty
+                content
+                likes
+                dislikes
+                stats
+            }
+        }
+        ''',
+        'variables': {'titleSlug': question_title_slug}
+
+    }
+
+    try:
+        response = requests.post(url, json=data, headers=headers, timeout=10)
+
+    except Exception as e:
+        logger.exception(
+            "file: embeds/question_embeds.py ~ Question could not be retrieved: %s", e)
+
+        embed = error_embed("Question could not be retrieved")
+        return embed
+
+    if response.status_code != 200:
+        logger.exception(
+            "file: embeds/question_embeds.py ~ Question could not be retrieved. Error code: %s", response.status_code)
+
+        embed = error_embed("Question could not be retrieved")
+        return embed
+
+    response_data = response.json()
+    question_difficulty = response_data['data']['question']['difficulty']
+    question_id = response_data['data']['question']['questionFrontendId']
+    question_title = response_data['data']['question']['title']
+    question_content = response_data['data']['question']['content']
+    question_link = f'https://leetcode.com/problems/{question_title_slug}'
+    question_stats = ast.literal_eval(response_data['data']['question']['stats'])
+    question_total_accepted = question_stats['totalAccepted']
+    question_total_submission = question_stats['totalSubmission']
+    question_ac_rate = question_stats['acRate']
+
+    rating_data = get_rating_data(question_title)
+
+    rating_text = "Doesn't exist"
```

Comment:
just set it to None by default, as in question embeds you don't ever show the rating if it doesn't have a rating. Then in question embed check if the rating is None rather than checking if it is "Doesn't exist"

Reply:
rename this variable to question_rating